.github/workflows/run-integ-tests.yml

```diff
@@ -0,0 +1,118 @@
+name: run-integ-tests
+
+on:
+  push: #TODO: remove this after finishing development
+    branches:
+      - ijemmy/integ-tests-on-aws
+  workflow_dispatch: {}
+
+env:
+  STACK_NAME: retaildemostore
+  STAGE_S3_BUCKET: retail-demo-store-integ-test-assets
+  REGION: us-west-2
+
+jobs:
+  # stage-assets:
+  #   name: Staging Asset
+  #   runs-on: ubuntu-20.04
+  #   permissions:
+  #     id-token: write # needed to interact with GitHub's OIDC Token endpoint.
+  #     contents: read
+  #   steps:
+  #     - uses: actions/checkout@v3
+  #       name: Checkout Repository
+  #     - name: Configure AWS credentials
+  #       uses: aws-actions/configure-aws-credentials@v1.6.1
+  #       with:
+  #         role-to-assume: ${{ secrets.AWS_ROLE_ARN_TO_ASSUME }}
+  #         role-session-name: GitHub-Action-Stage-Assets
+  #         aws-region: ${{ env.REGION }}
+  #     - name: Build and upload assets to an S3 bucket
+  #       run: |
+  #         ./stage.sh ${{ env.STAGE_S3_BUCKET }} --private-s3
+  # deploy-stacks:
+  #   name: Deploy Stacks
+  #   needs: stage-assets
+  #   runs-on: ubuntu-20.04
+  #   permissions:
+  #     id-token: write # needed to interact with GitHub's OIDC Token endpoint.
+  #     contents: read
+  #   steps:
+  #     - uses: actions/checkout@v3
+  #       name: Checkout Repository
+  #     - name: Configure AWS credentials
+  #       uses: aws-actions/configure-aws-credentials@v1.6.1
+  #       with:
+  #         role-to-assume: ${{ secrets.AWS_ROLE_ARN_TO_ASSUME }}
+  #         role-session-name: GitHub-Action-Deploy-Stack
+  #         aws-region: ${{ env.REGION }}
+  #     - name: Deploy all stacks to AWS
+  #       run: |
+  #         ./scripts/deploy-cloudformation-stacks.sh ${{ env.STAGE_S3_BUCKET }} ${{ env.REGION }} ${{ env.STACK_NAME }}
+  run-integ-tests:
+    name: Run Integration Tests
+    # needs: deploy-stacks
+    runs-on: ubuntu-20.04
+    permissions:
+      id-token: write # needed to interact with GitHub's OIDC Token endpoint.
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        service: 
+          - name: products
+            output_key: ProductsServiceUrl
+            env_var_name: PRODUCTS_API_URL
+          - name: users
+            output_key: UsersServiceUrl
+            env_var_name: USERS_API_URL
+          - name: orders
+            output_key: OrdersServiceUrl
+            env_var_name: ORDERS_API_URL
+          - name: recommendations
+            output_key: RecommendationsServiceUrl
+            env_var_name: RECOMMENDATIONS_API_URL
```

Comment:
I use matrix strategy to run all 4 tests in parallel. 

This makes it faster, and we can see result for each service a separated run. 

![image](https://user-images.githubusercontent.com/61999/171994503-a0515444-2d67-420e-b17e-466843497e28.png)